src/main/kotlin/com/atlassian/performance/tools/report/table/TabbedAuiTable.kt

```diff
@@ -0,0 +1,102 @@
+package com.atlassian.performance.tools.report.table
+
+internal class AuiTabbedTableFactory(
+    private val namedTables: List<Pair<AuiTable, String>>
+) {
+    fun create() = TabbedAuiTable(
+        tabs = namedTables
+            .mapIndexed { i, (table, name) -> AuiTableTab(name, i.toString(), table) }
+    )
+}
+
+internal class AuiTableFactory(
+    headers: List<String>,
+    private val rows: List<Map<String, Any>>
+) {
+    private val headers = headers.mapIndexed { i, h -> AuiTableHeader(i.toString(), h) }
```

Comment:
Constructors shouldn't do work. This mapping should be done either by `create` or injected via the constructor.